### ISSUE TITLE
[BUG] qp.ctrl(..., work_wires=...) drops outer Controlled work_wires under program capture

### ISSUE THREAD

**Issue by Qottmann** [PennyLaneAI/pennylane#10065]
> AI-assisted draft for tracking a bug found while adding capture decomposition tests in #10052.

### Expected behavior

When a decomposition rule calls `qp.ctrl(base, control=..., work_wires=..., work_wire_type="zeroed")`, capturing the rule via `qp.capture.make_plxpr` and round-tripping through `qp.tape.plxpr_to_tape` should preserve the outer `Controlled` op's `work_wires`.

### Actual behavior

The outer `Controlled` op's `work_wires` are dropped after capture/plxpr round-trip. Eager queuing keeps them.

### Additional information

- Reproduces with concrete wire labels (not just abstract/dynamic wires).
- Breaks `_test_decomposition_rule` under capture for templates whose decomposition rules use `qp.ctrl(..., work_wires=...)`, e.g. `OutSquare` and `SignedOutSquare`.
- Likely related to how `ControlledOp2` is encoded in the generic `operator` capture primitive: the jaxpr only records `n_ctrls` and the base op's wire lengths, with no slot for the outer `Controlled`'s ancilla `work_wires`.
- Currently xfailed in:
  - `tests/templates/subroutines/arithmetic/test_out_square.py::TestOutSquare::test_decomposition_rules_with_capture`
  - `tests/templates/subroutines/arithmetic/test_signed_out_square.py::TestSignedOutSquare::test_decomposition_rule_with_capture`

### Source code

```python
import pennylane as qp
import jax

def f(x_wires, output_wires, work_wires):
    base = qp.SemiAdder(x_wires, output_wires, work_wires[:1])
    qp.ctrl(
        base,
        control=work_wires[1:2],
        work_wires=work_wires[2:],
        work_wire_type="zeroed",
    )

kwargs = {"x_wires": [0, 1, 2, 3], "output_wires": [4, 5], "work_wires": [9, 10, 11]}

with qp.queuing.AnnotatedQueue() as q:
    f(**kwargs)
print("eager:  ", q.queue[0])

qp.capture.enable()
plxpr = qp.capture.make_plxpr(f, autograph=False)(**kwargs)
flat = jax.tree.leaves(((), kwargs))
tape = qp.tape.plxpr_to_tape(plxpr.jaxpr, plxpr.consts, *flat)
print("capture:", tape.operations[0])
qp.capture.disable()
```

Expected output difference:

```
eager:   Controlled(..., control_wires=[10], work_wires=[11])
capture: Controlled(..., control_wires=[10])  # work_wires missing
```

**Comment by anishbhujbal7** [comment]
Hi @PennyLaneAI team!

I would like to work on this issue.

The reproducer script clearly demonstrates the bug—when qp.ctrl operations with work_wires are captured via make_plxpr and converted back using plxpr_to_tape, the work_wires metadata is dropped during the primitive encoding step.

I plan to update the JAXPR primitive tracing for Controlled operations to preserve work_wires and update plxpr_to_tape to unpack them properly, enabling the current xfail tests in test_out_square.py and test_signed_out_square.py.

Could you please assign this issue to me? Thanks

**Comment by anishbhujbal7** [comment]
I was trying to resolve this issue and this are the things that I ran into
1. Work Wires Metadata Loss during JAXPR Capture

Issue: work_wires and work_wire_type metadata were lost during JAXPR program capture (qp.capture.enable()) for controlled Operator2 subroutines across both functional (ctrl_transform_prim) and class-based (ControlledOp2) paths.

Proposed Solution: Update controlled.py, controlled2.py, operator2.py, and base_interpreter.py to correctly pack work_wires and work_wire_type into primitive invars and params during JAXPR tracing, ensuring full metadata propagation when reconstructing tapes from PLXPR.

2. RecursionError in Test Setup (tests/conftest.py)

Issue: Running local tests without the optional pytest-rng plugin caused the seed fixture in conftest.py to look up itself and enter an infinite recursion loop.

Proposed Solution: Update the seed fixture lookup in conftest.py to explicitly filter out its own definition, safely falling back to a default seed (0) when pytest-rng is not installed.

3. Pybind11 Type Mismatch on Simulator Execution (lightning.qubit)
Issue: ControlledOp2.control_values stored control values as a NumPy ndarray, causing C++ execution failures in lightning.qubit which expects a standard Sequence[bool].
Proposed Solution: Update the control_values property getter in controlled2.py to convert internal array representations back to a Python tuple of booleans when accessed.

The fixes have been verified locally using a reproducer script and dedicated regression tests in test_templates.py

**Comment by Qottmann** [comment]
hey @anishbhujbal7 , this issue is critical to my work so I am already fixing it myself. Thanks for offering your help!

**Comment by anishbhujbal7** [comment]
No worries at all! Since I spent some time investigating it locally, I noticed a couple of edge cases that might be helpful while you're working on the fix:

JAXPR Metadata Loss: work_wires and work_wire_type are omitted from primitive invars/params in both ctrl_transform_prim (controlled.py) and ControlledOp2._bind_primitive (controlled2.py), causing loss during tape reconstruction in _op_impl (operator2.py).

Pybind11 Type Mismatch on C++ Backends: ControlledOp2.control_values returning a NumPy ndarray causes lightning.qubit execution to raise a TypeError expecting a standard Sequence[bool].
